[WIP] RHDEVDOCS 3910 and RHDEVDOCS-3902- Service binding discoverability and Topology usability improvements (#47760)
openshift/openshift-docs: modules/odc-creating-a-visual-connection-between-components.adoc

:_content-type: PROCEDURE
[id="odc-creating-a-visual-connection-between-components_{context}"]
== Creating a visual connection between components

> You can depict an intent to connect application components using the visual connector.

Reviewer: Please review the 21 instances of "using" in this PR and update them in accordance with the following IBM Style Guide guideline about "using":

To avoid ambiguity, replace this gerund with either "by using" or "that uses".
To refer to something "using" one thing or another is often ambiguous. For example, the sentence "Back up the server using
ABC" might mean "Back up the server by using ABC" or "Back up the server that uses ABC". This advice also applies to gerunds
other than "using". For more information, see the style and grammar tips in the Global audiences topic in IBM Style and the
section about "Clarity" in "Developing Quality Technical Information".

Author: I have replaced "using" gerund in this PR  with either "by using" or "that uses" where it is feasible. Thank you!

> Using the visual connector, you can depict an intent to connect application components.

Author: Reverting back to original content

This procedure walks you through an example of creating a visual connection between a PostgreSQL Database service and a Spring PetClinic sample application.

.Prerequisites

> * You have created and deployed a Spring PetClinic sample application that uses the *Developer* perspective.
> * You have created and deployed a Crunchy PostgreSQL database instance that uses the *Developer* perspective. This instance has the following four components: `hippo-backup`, `hippo-instance`, `hippo-repo-host`, and  `hippo-pgbouncer`.

Reviewer: ```suggestion
* You have created and deployed a Spring PetClinic sample application by using the *Developer* perspective.
* You have created and deployed a Crunchy PostgreSQL database instance by using the *Developer* perspective. This instance has the following components: `hippo-backup`, `hippo-instance`, `hippo-repo-host`, and  `hippo-pgbouncer`.
```

.Procedure

. Hover over the Spring PetClinic sample application to see a dangling arrow on the node.
+
.Visual connector
image::odc_connector.png[]
. Click and drag the arrow towards the `hippo-pgbouncer` deployment to connect the Spring PetClinic sample application with it.
. Click the `spring-petclinic` deployment to see the *Overview* panel. Under the *Details* tab, click the edit icon in the *Annotations* section to see the *Key = `app.openshift.io/connects-to`* and *Value = `[{"apiVersion":"apps/v1","kind":"Deployment","name":"hippo-pgbouncer"}]`* annotation added to the deployment.

> Similarly you can create other applications and components and establish visual connections between them.

Author: Done.

.Connecting multiple applications

> image::odc_connecting_multiple_applications.png[]

Author: Done.
